### DISCUSSION TITLE
How to stream large data from a RPC endpoint ?

### DISCUSSION THREAD

**Discussion by artech-git** [google/tarpc#445] [Q&A]
Let's say you have defined a service methods with the following methods 
```rust
#[tarpc::service]
pub trait BlockService{ 
    async fn stream_block_data(data: SomeData) -> SerializeStream<Bytes>; 
}
```
According to my research so far I didn't found any suitable way in which both the trait can be created ( serde::Serialize & futures::Stream) unless we plan to do it ourself( which is very tedious) 

based upon that given context can someone please tell me how to overcome this challenge, do we even need to consider it or is there something Iam missing from the piece of the puzzle, especially the internal working?

**Comment by tikue** [comment]
Hey, sorry, I didn't see this post earlier (I haven't used GitHub discussions before).

I am afraid streams are not supported by tarpc. I would suggest looking into an http2 server framework for streaming support.
